catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/util/AttributeHelper.java

```diff
@@ -0,0 +1,199 @@
+                .map(systemAndMetacardDescriptors::get)
+                .filter(Objects::nonNull)
+                .map(ad -> createAttribute(ad, attributeMap.get(ad.getName())))
+                .filter(Objects::nonNull)
+                .forEach(metacard::setAttribute));
+    }
+
+    /**
+     * Attempts to create an {@link Attribute} according to the provided {@link AttributeDescriptor}
+     * whose value is represented by the given string {@param parsedValue}. Returns null if the string
+     * could not be parsed based on the descriptor's underlying {@link ddf.catalog.data.AttributeType.AttributeFormat}.
+     *
+     * @param attributeDescriptor The descriptor holding the format to use for parsing.
+     * @param parsedValue         The value to be parsed.
+     * @return A new {@link Attribute} if parsing succeeded, or null otherwise.
+     */
+    public Attribute createAttribute(AttributeDescriptor attributeDescriptor, String parsedValue) {
+        try {
+            if (attributeDescriptor.isMultiValued()) {
+                LOGGER.trace("Found multi-valued attribute descriptor: {}",
+                        attributeDescriptor.getName());
+                return createMultiValuedAttribute(attributeDescriptor.getName(), parsedValue);
+            }
+
+            Serializable newValue;
+            switch (attributeDescriptor.getType()
+                    .getAttributeFormat()) {
+
+            case INTEGER:
+                newValue = Integer.parseInt(parsedValue);
+                break;
+
+            case FLOAT:
+                newValue = Float.parseFloat(parsedValue);
+                break;
+
+            case DOUBLE:
+                newValue = Double.parseDouble(parsedValue);
+                break;
+
+            case SHORT:
+                newValue = Short.parseShort(parsedValue);
+                break;
+            case LONG:
+                newValue = Long.parseLong(parsedValue);
+                break;
+
+            case DATE:
+                Calendar calendar = DatatypeConverter.parseDateTime(parsedValue);
+                newValue = calendar.getTime();
+                break;
+
+            case BOOLEAN:
+                newValue = Boolean.parseBoolean(parsedValue);
+                break;
+
+            case BINARY:
+                newValue = parsedValue.getBytes(Charset.forName("UTF-8"));
+                break;
+
+            case OBJECT:
+            case STRING:
+            case GEOMETRY:
+            case XML:
+                newValue = parsedValue;
+                break;
+
+            default:
+                return null;
+            }
+
+            return new AttributeImpl(attributeDescriptor.getName(), newValue);
+
+        } catch (IllegalArgumentException e) {
+            return null;
+        }
+    }
+
+    private void loadUniqueDescriptors(Set<AttributeDescriptor> descriptorsToLoad,
+            Map<String, AttributeDescriptor> destination) {
```

Comment:
one fun thing I have been doing (which idk if its against our "best practices" but whatevs) Is whenever I do have to do something mutable, I like to prefix my method param with `/*mutable*/` So the method declaration reads like
```java
private void loadUniqueDescriptors(Set<AttributeDescriptor> descriptorsToLoad,
            /*mutable*/ Map<String, AttributeDescriptor> destination) {
```
Helps me keep track of the places I want to go and destroy later ;) also helps notify the callers that you will be mutating some of their data they give you :)

Up to you though if you do or don't want to do this.